server/src/main/java/org/apache/druid/segment/indexing/DataSchema.java

```diff
@@ -123,6 +126,30 @@ public DataSchema(
     }
   }
 
+  public DataSchema(
+      @JsonProperty("dataSource") String dataSource,
```

Comment:
There should be only one constructor with `@JsonProperty` annotations.
Also, is it possible to use one of the existing constructors instead of adding a new one?

Reply:
Added the new constructor to handle the new field and retained the previous one to not have to change the heavily used signature everywhere -- primarily because this would be a temporary fix. Will remove the `@JsonProperty` part however.

Reply:
Have now moved the additional param to a new `CombinedDataSchema` class and reverted changes to `DataSchema`.

Reply:
Thanks!